test/Microsoft.AspNet.Mvc.IntegrationTests/ModelPrefixSelectionIntegrationTest.cs

```diff
@@ -0,0 +1,394 @@
+                request.QueryString = new QueryString("?Name=");
+            });
+
+            var modelState = new ModelStateDictionary();
+
+            // Act
+            var modelBindingResult = await argumentBinder.BindModelAsync(parameter, modelState, operationContext);
+
+            // Assert
+            Assert.NotNull(modelBindingResult);
+            Assert.True(modelBindingResult.IsModelSet);
+            Assert.Equal(string.Empty, modelBindingResult.Key);
+
+            var model = Assert.IsType<Person3>(modelBindingResult.Model);
+            Assert.Null(model.Name);
+
+            Assert.Equal(0, modelState.Count);
+            Assert.Equal(0, modelState.ErrorCount);
+            Assert.True(modelState.IsValid);
+        }
+
+        private class Person4
+        {
+            [FromForm]
+            public string Name { get; set; }
+        }
+
+        [Fact]
+        public async Task ComplexModel_EmptyPrefixSelected_NoMatchingValueProviderValue_WithFilteredValueProviders()
+        {
+            // Arrange
+            var argumentBinder = ModelBindingTestHelper.GetArgumentBinder();
+            var parameter = new ParameterDescriptor()
+            {
+                Name = "parameter",
+                ParameterType = typeof(Person4),
+                BindingInfo = new BindingInfo()
+                {
+                    BindingSource = BindingSource.Query,
+                },
+            };
+
+            var operationContext = ModelBindingTestHelper.GetOperationBindingContext(request =>
+            {
+                // This will only match empty prefix, but can't be used because of [FromForm] on the property.
+                request.QueryString = new QueryString("?Name=");
+
+                // This value won't be used to select a prefix, because we're only looking at the query string.
+                request.Form = new FormCollection(new Dictionary<string, StringValues>()
+                {
+                    { "parameter", string.Empty },
+                });
+            });
+
+            var modelState = new ModelStateDictionary();
+
+            // Act
+            var modelBindingResult = await argumentBinder.BindModelAsync(parameter, modelState, operationContext);
+
+            // Assert
+            Assert.NotNull(modelBindingResult);
+            Assert.True(modelBindingResult.IsModelSet);
+            Assert.Equal(string.Empty, modelBindingResult.Key);
+
+            var model = Assert.IsType<Person4>(modelBindingResult.Model);
+            Assert.Null(model.Name);
+
+            Assert.Equal(0, modelState.Count);
+            Assert.Equal(0, modelState.ErrorCount);
+            Assert.True(modelState.IsValid);
+        }
+
+        private class PurchaseOrder1
+        {
+            [FromHeader]
+            public string CustomerId { get; set; }
+        }
+
+        [Fact]
+        public async Task ComplexModel_WithGreedyProperties_PrefixSelectedByValueInValueProvider()
```

Comment:
Is anything at all different in the assertions between this group of tests and the `PersonX` ones? If yes, I suspect something is wrong.

Greedy properties should not be relevant since `IsTopLevelObject` will be `false` when they're checked. Would however see different behaviour with `parameter.BindingSource = BindingSource.Header;` Don't see any tests like that...